File: java/L100/LeetCode0137.java
package L100;

public class LeetCode0137 {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int i = 0;i < 32;++i) {
            int total = 0;
            for (int num : nums) {
                total += ((num >> i) & 1);
            }
            if (total % 3 != 0) {
                ans |= (1 << i);
            }
        }
        return ans;
    }
}

File: cpp/L100/LeetCode0137.cc
#include <vector>
using std::vector;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int a{0}, b{0};
        for (int num : nums) {
            b = ~a & (b ^ num);
            a = ~b & (a ^ num);
        }
        return b;
    }
};